### PR TITLE
Fix playwright result reporting for js-sdk upstream tests

### PR DESCRIPTION
**Pull request by dbkr**
Make the reporter job check out the react sdk explictly so it gets the right thing when used from other workflows like the js-sdk.

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: .github/workflows/end-to-end-tests.yaml [hunk 1]
```diff
@@ -133,6 +133,11 @@ jobs:
             deployments: write
         steps:
             - uses: actions/checkout@v4
+              with:
+                  ref: ${{ steps.sha.outputs.sha }}
```
Comment (t3chguy) [inline .github/workflows/end-to-end-tests.yaml:137]: You can't reference steps in other jobs, you'd have to make it an output in that job if that's what you wanted to achieve

File: .github/workflows/end-to-end-tests.yaml [hunk 2]
```diff
@@ -133,6 +133,11 @@ jobs:
...
+                  persist-credentials: false
+                  path: matrix-react-sdk
```
Comment (t3chguy) [inline .github/workflows/end-to-end-tests.yaml:139]: checking out in a subdirectory would mean all later steps need to cd into it

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: .github/workflows/end-to-end-tests.yaml (hunk 1 above)
Comment (dbkr) [reply .github/workflows/end-to-end-tests.yaml]: I jst copied it from above so probably cargoculting - I guess this and the other one are just unnecessary?

File: .github/workflows/end-to-end-tests.yaml (hunk 2 above)
Comment (dbkr) [reply .github/workflows/end-to-end-tests.yaml]: Yeah, presumably just using the default dir is fine?

Comment (t3chguy) [reply .github/workflows/end-to-end-tests.yaml]: Yup!

File: .github/workflows/end-to-end-tests.yaml (hunk 1 above)
Comment (t3chguy) [reply .github/workflows/end-to-end-tests.yaml]: the other one is necessary, if your PR is not based off latest you don't want to run the tests for the default branch, you want to run the tests you wrote in your PR.

**Review by t3chguy — APPROVED** [review]

**Pull request merged**